boards/p1am_100/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+#![no_std]
+
+extern crate atsamd_hal as hal;
+
+#[cfg(feature = "rt")]
+extern crate cortex_m_rt;
+#[cfg(feature = "rt")]
+pub use cortex_m_rt::entry;
+
+use hal::prelude::*;
+use hal::*;
+
+pub use hal::common::*;
+
+pub use hal::target_device as pac;
+
+use gpio::v2::AnyPin;
+use gpio::Port;
+use hal::clock::GenericClockController;
+use hal::sercom::v2::spi;
+use hal::sercom::{I2CMaster0, PadPin, UART5};
+use hal::time::Hertz;
+
+#[cfg(feature = "usb")]
+use hal::usb::usb_device::bus::UsbBusAllocator;
+#[cfg(feature = "usb")]
+pub use hal::usb::UsbBus;
+
+bsp_pins!(
```

Comment:
While porting the feather_m0, I realized that this macro needed some tweaking. I added the ability to provide doc strings on the names as well as the aliases.